CMakeLists.txt

```diff
@@ -13,6 +13,8 @@ FATAL: In-source builds are not allowed.
 ")
 endif()
 
+# Useful for HALs if built as a CMake submodule
+set(OPENCV_SOURCE_DIR ${CMAKE_CURRENT_SOURCE_DIR})
```

Comment:
There is CMake's `<project>_SOURCE_DIR` for that: https://cmake.org/cmake/help/latest/variable/PROJECT-NAME_SOURCE_DIR.html

Should be used as `OpenCV_SOURCE_DIR`.